File: python/examples/market_order.py
#!/usr/bin/env python3
"""
Market Order Example

Place a market order that executes immediately at best available price.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python market_order.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid Market Order Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Address: {sdk.address}")

    # Market buy by notional ($11 worth of BTC - minimum is $10)
    order = sdk.market_buy("BTC", notional=11)
    print(f"Market buy: {order}")
    print(f"  Status: {order.status}")
    print(f"  OID: {order.oid}")

    # Market buy by notional ($10 worth of ETH)
    # order = sdk.market_buy("ETH", notional=10)

    # Market sell
    # order = sdk.market_sell("BTC", size=0.0001)


if __name__ == "__main__":
    main()

File: typescript/examples/market_order.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Market Order Example
 *
 * Place a market order that executes immediately at best available price.
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });

  // Market buy by notional ($11 worth of BTC - minimum is $10)
  const order = await sdk.marketBuy("BTC", { notional: 11 });
  console.log(`Market buy: ${order}`);
  console.log(`  Status: ${order.status}`);
  console.log(`  OID: ${order.oid}`);

  // Market buy by notional ($10 worth of ETH)
  // const order = await sdk.marketBuy("ETH", { notional: 10 });

  // Market sell
  // const order = await sdk.marketSell("BTC", { size: 0.0001 });
}

main().catch(console.error);
